hummingbot/connector/exchange/gate_io/gate_io_exchange.py

```diff
@@ -420,14 +420,17 @@ def _process_trade_message(self, trade: Dict[str, Any], client_order_id: Optiona
             self.logger().debug(f"Ignoring trade message with id {client_order_id}: not in in_flight_orders.")
             return
 
+        # When GT token is used to pay fee 'gt_fee' is provided by the API
+        if 'gt_fee' in trade and Decimal(trade["gt_fee"]) > 0:
+            flat_fees = [TokenAmount(amount=Decimal(trade["gt_fee"]), token='GT')]
+        else:
+            flat_fees = [TokenAmount(amount=Decimal(trade["fee"]), token=trade["fee_currency"])]
+
```

Comment:
The API documentation does not discard the case where fee is paid both in GT and in another token. We should prepare the logic for that scenario, by adding a flat fee for GT fee (if there is any GT fee) and a flat fee for other token fee (if there is and it is not 0).

-- Reviewed in representation of Hummingbot Foundation Technical DAO

Reply:
Gate.io support had replied to my ticket related to this, that if I set the 'Pay with GT' flag, I cannot use my points, thus I did not think about that. Thanks for pointing that out, in Gate.io case, this also requires to check and add the 'point_fee'
Now the display will be tricky as fee is only displayed in 1 currency or should this feature be also supported by the 'history' output?

Reply:
Hi @aarmoa @MementoRC,

Today we discuss this issue in the Hummingbot Foundation Meeting and we have a proposal for this situation, but we would like to let the decision open to the community.
We think that the fees has to admit payments in the native token of the exchanges (like GT in this case) or base/quote, but POINTS are only used in Gate.io so we think we don't need to include it.